{{cookiecutter.project_shortname}}/webpack.config.js

```diff
@@ -3,11 +3,11 @@ const packagejson = require('./package.json');
 
 const dashLibraryName = packagejson.name.replace(/-/g, '_');
 
-module.exports = {
+module.exports = (env, argv) => ({
     entry: {main: './src/lib/index.js'},
     output: {
         path: path.resolve(__dirname, dashLibraryName),
-        filename: 'bundle.js',
+        filename: `${dashLibraryName}.${argv.mode === 'development' ? 'dev' : 'min'}.js`,
```

Comment:
We may not be able to assume `argv` will have a `mode` field... or even that `argv` will exist. I have matched the configs of dash-renderer to this PR and when running `npm run start` (webpack-serve runs the config for us) I get the error 

```
➜  dash-renderer git:(master) ✗ npm run start

> dash-renderer@0.14.1 start /home/ben/plotly/dash-renderer
> webpack-serve ./webpack.serve.config.js --open --mode development

/home/ben/plotly/dash-renderer/webpack.config.js:10
        filename: `${dashLibraryName}.${argv.mode === 'development' ? 'dev' : 'min'}.js`,
                                             ^

TypeError: Cannot read property 'mode' of undefined
    at module.exports (/home/ben/plotly/dash-renderer/webpack.config.js:10:46)
    at fromFunction (/home/ben/plotly/dash-renderer/node_modules/@webpack-contrib/config-loader/lib/resolve.js:4:18)
    at module.exports (/home/ben/plotly/dash-renderer/node_modules/@webpack-contrib/config-loader/lib/resolve.js:24:10)
    at module.exports (/home/ben/plotly/dash-renderer/node_modules/@webpack-contrib/config-loader/lib/index.js:17:10)
    at load (/home/ben/plotly/dash-renderer/node_modules/webpack-serve/lib/config.js:107:12)
    at module.exports (/home/ben/plotly/dash-renderer/node_modules/webpack-serve/lib/options.js:65:10)
    at module.exports (/home/ben/plotly/dash-renderer/node_modules/webpack-serve/index.js:16:10)
    at Object.<anonymous> (/home/ben/plotly/dash-renderer/node_modules/webpack-serve/cli.js:82:1)
    at Module._compile (internal/modules/cjs/loader.js:689:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! dash-renderer@0.14.1 start: `webpack-serve ./webpack.serve.config.js --open --mode development`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the dash-renderer@0.14.1 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/ben/.npm/_logs/2018-10-02T14_42_14_950Z-debug.log
```

Reply:
In this case webpack-server has already set mode for us - the module.exports objecti s already set.

Here is an example solution:

```
const path = require('path');
const packagejson = require('./package.json');

const dashLibraryName = packagejson.name.replace(/-/g, '_');

module.exports = function(env, argv) {

    let mode;

    // priority mode is set by flags
    if (argv && argv.mode) {
        mode = argv.mode;
    }

    // secondary mode setting by examining existing configuration object (module.exports)
    else if (module.exports && module.exports.mode) {
        mode = module.exports = mode;
    }

    // else take webpack default
    else {
        mode = 'production'; // webpack default
    }


    return {
        entry: {main: './src/index.js'},
        output: {
            path: path.resolve(__dirname, dashLibraryName),
            filename: mode === 'development' ? `${dashLibraryName}.dev.js` : `${dashLibraryName}.min.js`,
            library: dashLibraryName,
            libraryTarget: 'window',
        },
        externals: {
            react: 'React',
            'react-dom': 'ReactDOM',
            'plotly.js': 'Plotly',
        },
        module: {
            rules: [
                {
                    test: /\.js$/,
                    exclude: /node_modules/,
                    use: {
                        loader: 'babel-loader',
                    },
                }
            ],
        },
        devtool: mode === 'development' ? "eval-source-map" : 'none'
    };
};
```

Reply:
Good point, I always use `build:js` and `build:js-dev` so I didn't get to see the error (I don't like webpack watcher, I use pycharm watch instead). I think it would have to be fixed in the other repos too.